### Урок 1. Случайные события. Условная вероятность. Формула Байеса. Независимые испытания#

### -- Автор: Шенк Евгений Станиславович

In [1]:
import math
import numpy as np

### Задание 1.

Из колоды в 52 карты вынимают случайным образом 4 карты. Найти число исходов, соответствующих тому, что был вытянут хотя бы один туз.

A - число исходов, соответствующих тому, что был вытянут хотя бы один туз

Найдем A как разность между числом исходов при всех 52 картах и исходах без тузов (48 карт)

$$C_{52}^4 = \frac{52!}{4! \: (52 - 4)!} = \frac{52!}{4! \: 48!} = \frac{49 \cdot 50 \cdot 51 \cdot 52}{4!} = \frac{49 \cdot 50 \cdot 51 \cdot 52}{2 \cdot 3 \cdot 4} = 270725$$

$$C_{48}^4 = \frac{48!}{4! \: (48 - 4)!} = \frac{48!}{4! \: 44!} = \frac{45 \cdot 46 \cdot 47 \cdot 48}{4!} = \frac{45 \cdot 46 \cdot 47 \cdot 48}{2 \cdot 3 \cdot 4} = 194580$$

$$A = 270725 - 194580 = 76145$$

In [2]:
def combinations(n: int, k: int) -> int:
    """Число сочетаний.
    """
    
    return np.math.factorial(n) // (np.math.factorial(k) * np.math.factorial(n - k))

In [3]:
A = combinations(52, 4) - combinations(48, 4)
A

76145

Ответ: 76145

### Задание 2.

Семь человек рассаживаются наудачу на скамейке. Какова вероятность того, что рядом будут сидеть: а) два определённых человека? б) три определённых человека?

$$P(A) = \dfrac{m}{n}.$$

In [4]:
n = math.factorial(7)
n

5040

количество исходов всего: n = 5040

а) два определённых человека

могут меняться между собой местами и занять одну из 6 позиций (1-2, 2-3, 3-4, 4-5, 5-6, 6-7)

m1 = 2! * 6 = 12 - количество успешных исходов

$$P(A) = \dfrac{m1}{n} = \dfrac{12}{5040} = 0,002381.$$

б) три определённых человека

могут меняться между собой местами и занять одну из 5 позиций (1-2-3, 2-3-4, 3-4-5, 4-5-6, 5-6-7)

m2 = 3! * 5 = 6 * 5 = 30 - количество успешных исходов

$$P(A) = \dfrac{m2}{n} = \dfrac{30}{5040} = 0,005952.$$

### Задание 3.

Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. Какова вероятность того, что среди трёх наугад выбранных вопросов студент знает: а) все? б) два?

In [5]:
def arrangements(n: int, k: int) -> int:
    """Число размещений.
    """
    
    return np.math.factorial(n) // np.math.factorial(n - k)

а) Знает все?

$$P(A) =\dfrac{50}{60} \cdot \dfrac{49}{59} \cdot \dfrac{48}{58}  = \dfrac{117600}{205320} = 0.5728$$

После каждого вопроса общее количество и количество выученных вопросов уменьшается на 1

In [6]:
arrangements(50, 3) / arrangements(60, 3)

0.5727644652250146

Ответ: 0,5728

б) Знает два?

$$P(A) =\dfrac{50}{60} \cdot \dfrac{49}{59} \cdot \dfrac{10}{58}  = \dfrac{24500}{205320} = 0.1193$$

Два раза берем количество успехов и один раз неудачи (60 - 50 = 10)

In [7]:
arrangements(50, 2) * arrangements(10, 1) / arrangements(60, 3)

0.11932593025521138

Ответ: 0,1193

### Задание 4.

Бросается игральная кость. Пусть событие A - появление чётного числа, событие B - появление числа больше трёх. Являются ли эти события независимыми?

$P(A) = 1/2$ (три числа (2, 4, 6) из шести)

$P(B) = 1/2$ (три числа (4, 5, 6) из шести)

Посчитаем совместную вероятность $P(AB)$. Событие $AB$ соответствует выбору числа 4 или 6, т.е. $P(AB) = 2/6 = 1/3$. Следовательно, $$P(A|B) = \dfrac{P(AB)}{P(B)} = \dfrac{\frac{1}{3}}{\frac{1}{2}} = \dfrac{2}{3}.$$

Поскольку $P(A|B) \neq P(A); \dfrac{2}{3} \neq \dfrac{1}{2}$, заключаем, что события $A$ и $B$ зависимы.

Ответ: не являются независимыми

### Задание 5.

Допустим, имеется некоторая очень редкая болезнь (поражает 0.1% населения). Вы приходите к врачу, вам делают тест на эту болезнь, и тест оказывается положительным. Врач говорит вам, что этот тест верно выявляет 99% больных этой болезнью и всего лишь в 1% случаев даёт ложный положительный ответ.

Вопрос: какова вероятность, что вы действительно больны ей?

Подсказка: вновь используйте формулу Байеса с раскрытием знаменателя с помощью формулы полной вероятности.

$$P(A|B) = \dfrac{P(B|A) \cdot P(A)}{P(B)}$$

$$P(B) = P(B | H_1) \cdot P(H_1) + P(B | H_2) \cdot P(H_2) + \dots + P(B | H_n) \cdot P(H_n).$$

P(B|H_1) - вероятность заболевания: 0,001

P(B|H_2) - вероятность что здоров: 0,999

P(A) - действительно болен: 0,001

P(B) - вероятность быть признаным больным (рассчитаем полную вероятность)

P(B|A) - вероятность корректного определения больного: 0,99

$$P(B) = 0.001 \cdot 0.99 + 0.999 \cdot 0.01 = 0.01098.$$

$$P(A|B) = \dfrac{0.001 \cdot 0.99}{0.001 \cdot 0.99 + 0.999 \cdot 0.01} = 0.090164.$$

Ответ: вероятность, что вы действительно больны ей всего около 9%